src/inmanta/plugins.py

```diff
@@ -219,6 +225,78 @@ def type_string_internal(self) -> str:
     None: Null(),  # Only NoneValue will pass validation
 }
 
+python_to_model = {
+    str: inmanta_type.String(),
+    float: inmanta_type.Float(),
+    numbers.Number: inmanta_type.Number(),
+    int: inmanta_type.Integer(),
+    bool: inmanta_type.Bool(),
+    dict: inmanta_type.LiteralDict(),
+    list: inmanta_type.LiteralList(),
+}
+
+
+def to_dsl_type(python_type: type[object]) -> inmanta_type.Type:
+    """
+    :param python_type: The evaluated python type as provided in the Python type annotation.
+    """
+    # Any to any
+    if python_type is typing.Any:
+        return inmanta_type.AnyType()
+
+    if python_type is type(None):
+        return Null()
+
+    if typing_inspect.is_union_type(python_type):
+        if any(typing_inspect.is_optional_type(tt) for tt in typing.get_args(python_type)):
+            # Optional type
+            other_types = [tt for tt in typing.get_args(python_type) if not typing_inspect.is_optional_type(tt)]
+            if len(other_types) == 0:
+                # Probably not possible
+                return Null()
+            if len(other_types) == 1:
+                return inmanta_type.NullableType(to_dsl_type(other_types[0]))
+            # TODO: optional unions
+            return inmanta_type.AnyType()
+        else:
+            return inmanta_type.AnyType()
+
+    # TODO: unions
+    # bases: Sequence[inmanta.ast.type.Type] = [to_dsl_type(arg) for arg in typing.get_args(python_type)]
+    # return inmanta.ast.type.Union(bases)
+
+    if typing_inspect.is_generic_type(python_type):
+        origin = typing.get_origin(python_type)
+        if  origin is Sequence:
+            # Can this fail?
```

Comment:
Question to reviewers

Reply:
Is it the same at collections.abc.Sequence? And does this support `list[str]`?

Reply:
@wouterdb as to your question: I think it can indeed fail for implicit `Any`. See below
```
>>> from collections.abc import Sequence
>>> import typing
>>> typing.get_args(Sequence)
()
```

> Is it the same at collections.abc.Sequence?

It should be that one really. The import from `typing` is deprecated.

> And does this support list[str]?

I don't think so. I also had a comment about this.